File: hackerearth/Fav Singer.cpp
#include <iostream>
using namespace std;

int main() {
    long n;
    cin >> n;
    
    long arr[n];
    long brr[1000000] = {0};

    for (int i = 0; i < n; i++) {
        cin >> arr[i];
        brr[arr[i] - 1]++;
    }

    long max = 0;
    for (int i = 0; i < 1000000; i++) {
        if (brr[i] > max) {
            max = brr[i];
        }
    }

    long count = 0;
    for (int i = 0; i < 1000000; i++) {
        if (brr[i] == max) {
            count++;
        }
    }

    cout << count;
    return 0;
}

File: hackerearth/Fav Singer.java
import java.util.*;
import java.util.stream.Collectors;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt(); 
        long[] singers = new long[n];

        for (int i = 0; i < n; i++) {
            singers[i] = scanner.nextLong();  
        }
        Map<Long, Long> singerFrequency = Arrays.stream(singers)
                .boxed()
                .collect(Collectors.groupingBy(singer -> singer, Collectors.counting()));

        long maxFrequency = singerFrequency.values().stream()
                .mapToLong(Long::longValue)
                .max()
                .orElse(0);

        long countFavoriteSingers = singerFrequency.values().stream()
                .filter(frequency -> frequency == maxFrequency)
                .count();

        System.out.println(countFavoriteSingers);
    }
}
